### DISCUSSION TITLE
How to get outputdevice on MacOS?

### DISCUSSION THREAD

**Discussion by garcon33** [melanchall/drywetmidi#182] [Q&A] [answered]
Hello,

I cannot detect any Midi devices on MacOS. 
Would you give me some tips to get them?

I checked the number of detected devices with `OutputDevice.GetDevicesCount()`, and it throws 0.

I just want to play sounds from embedded speakers or headphones.
I'm developing the program using Unity, and it can detect midi devices in Windows environment.

Many thanks in advance,
Yj

**Comment by melanchall** [comment, marked as answer]
Hello,

What version of the library you use? What devices standard macOS tools list (for example, what do you see here: In the Audio MIDI Setup app on your Mac, choose Window > Show MIDI Studio)?

Also please note that embedded speakers or headphones are **not MIDI devices**. So they won't be listed as MIDI devices both by library and by macOS at all.

**Reply by garcon33** [reply]
Thanks for quick replies!

The version of the library is 6.0.1.
I see only "IAC driver" on the MIDI Studio.

Yj

**Reply by garcon33** [reply]
Hello again,

I could detect "IAC driver bus1" by activating the checkbox "Device is online".
Yet, no sounds are played from the speakers...

Yj

**Reply by melanchall** [reply]
IAC Driver is built-in virtual MIDI device. Yes, if you make it online, DryWetMIDI will recognize it. All you see is totally correct. IAC Driver is just virtual cable. You can open some software synth, assign IAC Driver as input device, and in your program send data to output device with the same name. And now your data will be directed through this virtual cable to synth.

**Reply by garcon33** [reply]
Thanks for your explanation!
Now I can hear sounds via garageBand!
